src/Features/Core/Portable/UseCollectionInitializer/AbstractUseCollectionInitializerDiagnosticAnalyzer.cs

```diff
@@ -159,12 +159,10 @@ private void AnalyzeNode(SyntaxNodeAnalysisContext context, INamedTypeSymbol ien
                     context.ReportDiagnostic(Diagnostic.Create(
                         UnnecessaryWithSuggestionDescriptor, location1, additionalLocations: locations));
 
-                    context.ReportDiagnostic(Diagnostic.Create(
-                        UnnecessaryWithoutSuggestionDescriptor,
+                    context.ReportDiagnostic(CreateUnnecessaryWithoutSuggestionDiagnostic(
                         Location.Create(syntaxTree, TextSpan.FromBounds(
                             arguments.Last().FullSpan.End,
-                            match.Span.End)),
-                        additionalLocations: locations));
```

Comment:
Here's a case brought to light by my base class change where `additionalLocations` was being specified erroneously (it serves no purpose here).